src/lib/discord/commando/commands/autocreate.js

```diff
@@ -63,26 +71,218 @@ exports.run = async (client, msg, [args]) => {
 
 		const templateName = args.shift()
 
-		const template = channelTemplate.find((x) => x.name.toLowerCase() === templateName)
+		const template = channelTemplate.find((x) => x.name === templateName)
 		if (!template || !template.definition) {
 			return await msg.reply('I can\'t find that channel template! (remember it has to be your first parameter)')
 		}
 
 		// switch underscores back in so works for substitution later
+    const subArgs = []
 		for (let x = 0; x < args.length; x++) {
-			args[x] = args[x].replace(/ /g, '_')
+			subArgs[x] = args[x].replace(/ /g, '_')
 		}
 
 		let categoryId
 		if (template.definition.category) {
-			const category = await guild.channels.create(format(template.definition.category, args), { type: 'GUILD_CATEGORY' })
+			const categoryOptions = {
+				type: 'GUILD_CATEGORY',
+			}
+
+			const categoryName = format(template.definition.category.categoryName, args)
+
+			// add role permissions
+			let roleId
+			if (template.definition.category.roles) {
+				const roleOverwrites = []
+				for (const role of template.definition.category.roles) {
+					const allowed = []
+					const deny = []
+          const roleNames = guild.roles.cache.map(r => r.name)
+          const roleIds = guild.roles.cache.map(r => r.id)
+          for (let x = 0; x < roleNames.length; x++) {
+            if ((format(role.name, args)) === roleNames[x]) {
+              roleId = await guild.roles.cache.get(roleIds[x])
+            }
+          }
+          if (!roleId) {
+            roleId = await guild.roles.create({
+              name: (format(role.name, args)),
+              permissions: []
+            })
+          }
+					if (role.view) {
```

Comment:
these all default to false if not specified - is this a good idea?
If not you can do role.view ?? true - which would be true if they don't appear in the config, but would be false if specified in the config file
(I have not considered enough if it is... or if it does not appear is the lack of allowed or deny the right thing?

Reply:
I simply replicated and expanded on the existing logic. It seemed fine and hasn't caused any errors in my testing, so I don't see it being a problem.